app/controllers/homepage_controller.rb

```diff
@@ -21,7 +21,9 @@ def index
     listing_shape_menu_enabled = all_shapes.size > 1
     @show_categories = @categories.size > 1
     show_price_filter = @current_community.show_price_filter && all_shapes.any? { |s| s[:price_enabled] }
-    @show_custom_fields = @current_community.custom_fields.any?(&:can_filter?) || show_price_filter
+
+    filters = @current_community.custom_fields.where(search_filter: true).sort
```

Comment:
`sort` uses the default `<=>` method. So in this case the `sort` doesn't affect to the SQL query, instead it's just a normal sorting operation on the Ruby side.

The `CustomField` models includes `SortableByPriority` module, which implements the `<=>` method like this:

```ruby
  # Compare two object.
  #
  # Usage:
  # - Include this module to your ActiveRecord Model
  # - Add `sort_priority` to the model
  #
  # Returns 0, if objects are equal, otherwise by sort priority
  def <=> other
    (sort_priority || 0) <=> (other.sort_priority || 0)
  end
```